frontend/src/helpers/utils.js

```diff
@@ -47,41 +47,58 @@ export function getSimpleEquation(reaction) {
   return `${reactants} ${equationSign(reaction.reversible)} ${products}`;
 }
 
-export function getChemicalReaction(reaction) {
-  if (reaction === null) {
-    return '';
-  }
-  const reactants = reaction.reactants.map(
-    x => `${x.stoichiometry !== 1 ? `${x.stoichiometry} ` : ''}${x.fullName}`
-  ).join(' + ');
-  const products = reaction.products.map(
-    x => `${x.stoichiometry !== 1 ? `${x.stoichiometry} ` : ''}${x.fullName}`
-  ).join(' + ');
-  return `${reactants} ${equationSign(reaction.reversible)} ${products}`;
-}
 
 const sortByName = metabolites => [...metabolites].sort((a, b) => ((a.name > b.name) ? 1 : -1));
 
+/** Get the compartement from a reactant or product */
+const getCompartment = component => component.fullName.match(/\[[a-z]{1,3}\]/)[0];
+
+
+/** Create  the compartements for the summary, as used in Equation and Related Reactions */
+export const formatCompartmentStr = (reaction) => {
+  const reactants = reaction.metabolites.filter(m => m.outgoing);
+  const products = reaction.metabolites.filter(m => !m.outgoing);
+
+  /** Extract the compartements from the full names,
+    * discard duplicates and return as a string  */
+  function uniqueCompartments(xs) {
+    return Array.from(new Set(xs.map(r => getCompartment(r)))).join(' + ');
+  }
```

Comment:
Personally I would prefer to have this defined outside (and use the `const = uniqueCompartments = (xs) => ...` syntax) so it could be reused in other places too. I'm not sure if there is any best practice guidelines about this though.  I know we have existing examples of nested function definitions already though, maybe it's something worth discussing about as a group.

Reply:
I initially put this outside, but then thought there might be practice not to, as it's not used anywhere else (yet), and I noticed other inner functions in the code. But if you think it's better on its own, then I'll move it!
Also, if we have any guidelines about when to use `const` and when to use `function`, that would be nice.

Reply:
@e0 @MalinAhlberg should we create an issue regarding discussing/documenting const vs function?

Reply:
yes, or should it go as a subitem on the linting card?

Reply:
@MalinAhlberg good idea!